app/src/main/java/com/example/poste/activities/LoginActivity.java

```diff
@@ -1,99 +1,127 @@
 package com.example.poste.activities;
 
+import static com.example.poste.utils.ValidationUtils.validateEmail;
+
+import androidx.appcompat.app.AppCompatActivity;
+
 import android.content.Intent;
 import android.os.Bundle;
-import android.view.Window;
-import android.view.WindowManager;
+import android.text.SpannableString;
+import android.text.method.LinkMovementMethod;
+import android.text.style.ClickableSpan;
+import android.util.Log;
+import android.view.View;
 import android.widget.Button;
 import android.widget.EditText;
-import android.widget.Switch;
+import android.widget.TextView;
 import android.widget.Toast;
 
-import androidx.appcompat.app.AppCompatActivity;
-
-import com.example.poste.PosteApplication;
 import com.example.poste.R;
-import com.example.poste.api.poste.API;
-import com.example.poste.api.poste.exceptions.APIException;
-import com.example.poste.api.poste.models.User;
+import com.example.poste.http.LoginRequest;
+import com.example.poste.http.MyApiService;
+import com.example.poste.http.RetrofitClient;
+import com.example.poste.models.User;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
-/**
- * The LoginActivity class adds functionality to the activity_login.xml layout
- */
 public class LoginActivity extends AppCompatActivity {
+
     public Button buttonLoginSubmit;
-    private Switch rememberMeSwitch;
-    private EditText usernameField,passwordField;
-    private String email;
-    private String password;
-    private User loginUser;
+    private EditText usernameField, passwordField;
 
     /**
-     * Called when the activity is created
-     *
-     * @param savedInstanceState A bundle containing the saved instance state
+     * function that runs on creation of the activity
+     * @param savedInstanceState
      */
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
-        // Configure window settings for fullscreen mode
-        requestWindowFeature(Window.FEATURE_NO_TITLE);
-        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
-                WindowManager.LayoutParams.FLAG_FULLSCREEN);
-        getSupportActionBar().hide();
-
-        // Set the activity layout
         setContentView(R.layout.activity_login);
 
-        // Prep vars
-        usernameField = findViewById(R.id.LoginEmailEntertxt);
-        passwordField = findViewById(R.id.LoginPassword);
-        rememberMeSwitch = findViewById(R.id.login_remember_me_switch);
-
-        //Email = findViewById(R.poste_item_id.LoginEmailEntertxt);
+        TextView hyperlinkTextView = findViewById(R.id.hyperlinkTextViewToRegister);
+        usernameField = findViewById(R.id.editTextTextEmailAddress);
```

Comment:
Some of the var names seem a little redundant; perhaps a typo? Might help clarity to rename, but not a blocker.

Reply:
That was the naming scheme I was kind of going for, editText  is the widget, Text is the data type and EmailAddress is what it contains. Can be fixxed in the future. Needs changes in too many places to fix right now

Reply:
No worries. If that's what works for you, right on; not trying to pick apart coding style, just wanted to make sure there wasn't a typo :)